.github/workflows/ci.yaml

```diff
@@ -0,0 +1,61 @@
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+
+env:
+  INSTALL_MDBOOK: |
+    mkdir bin
+    curl -sSL https://github.com/rust-lang/mdBook/releases/download/v0.4.35/mdbook-v0.4.35-x86_64-unknown-linux-gnu.tar.gz \
+    | tar -xz --directory=bin
+  RUSTFLAGS: "-Dwarnings"
+
+jobs:
+  check_book:
+    name: Check mdbook
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: "$INSTALL_MDBOOK"
+      - run: bin/mdbook build
+
+  formatting:
+    name: Check mdformat
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: pip install mdformat-gfm
+      - run: mdformat --check --wrap 100 .
+
+  test:
+    runs-on: ubuntu-latest
+    name: Test Rust code
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@stable
+      - uses: Swatinem/rust-cache@v2
+      - run: |
+          cd test-book
+          cargo check --all-targets
+          cargo test
+
+  deploy:
+    name: Deploy book
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: "$INSTALL_MDBOOK"
+      - run: |
+        bin/mdbook build
+        git worktree add gh-pages
+        git config user.name "Deploy from CI"
+        git config user.email ""
+        cd gh-pages
+        # Delete the ref to avoid keeping history.
+        git update-ref -d refs/heads/gh-pages
+        rm -rf *
+        mv ../book/* .
+        git add .
+        git commit -m "Deploy $GITHUB_SHA to gh-pages"
+        git push --force --set-upstream origin gh-pages
```

Comment:
Hmm, seems it doesn't like this syntax?

https://github.com/RustCrypto/book/actions/runs/6386361630

> *Invalid workflow file*: .github/workflows/ci.yaml#L50
> You have an error in your yaml syntax on line 50

Reply:
I think I missed the indentation https://github.com/RustCrypto/book/pull/4